packages/core/http/core-http-server-internal/src/http_service.ts

```diff
@@ -222,12 +225,65 @@ export class HttpService
         await this.httpsRedirectServer.start(config);
       }
 
+      if (config.oas.enabled) {
+        this.log.info('Registering experimental OAS API');
+        this.registerOasApi(config);
+      }
+
       await this.httpServer.start();
     }
 
     return this.getStartContract();
   }
 
+  private registerOasApi(config: HttpConfig) {
```

Comment:
@jloleysens adding this here because I don't know where else it should go. When running this PR locally to test out the "how to test" section, jq throws when using a basepath.
You might want to give other reviewers the head's up to run kibana locally _without_ a base path.